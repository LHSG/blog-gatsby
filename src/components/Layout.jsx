import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Header from './Header';
import Footer from './Footer';

const LayoutDiv = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
`;

const ContentDiv = styled.div`
  margin: 3rem auto;
  max-width: 740px;
`;

export default function Layout({ path, children }) {
  return (
    <LayoutDiv>
      <Header path={path} />
      <ContentDiv>{children()}</ContentDiv>
      <Footer />
    </LayoutDiv>
  );
}

Layout.propTypes = {
  path: PropTypes.string.isRequired,
  children: PropTypes.func.isRequired,
};
