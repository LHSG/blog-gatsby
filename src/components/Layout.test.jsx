import React from 'react';
import { shallow } from 'enzyme';
import Layout from './Layout';

test('rendering', () => {
  const wrapper = shallow(
    <Layout path="/something">{() => <span>Some content</span>}</Layout>,
  );
  expect(wrapper).toMatchSnapshot();
});
