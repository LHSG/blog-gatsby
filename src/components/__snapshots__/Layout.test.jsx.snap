// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<styled.div>
  <Header
    path="/something"
  />
  <styled.div>
    <span>
      Some content
    </span>
  </styled.div>
  <Footer />
</styled.div>
`;
